components/amount-buttons.jsx

```diff
@@ -88,9 +88,7 @@ var AmountButtons = React.createClass({
     });
   },
   otherRadioChange: function() {
-    if (!this.state.userInputting) {
-      this.setAmount("", true);
-    }
+    this.setAmount("", true);
```

Comment:
If we ensure onRadioChange only fires if it goes from not checked to checked, we can remove the if statement above. Better off to push that logic into the other buttons itself, because it's only needed in the other buttons.